design/UsageValidationRules.md

```diff
@@ -0,0 +1,71 @@
+## Definition of "resources being used"
+
+Resources in WebGPU can be used in different ways that are declared at resource creation time and interact with various validation rules.
+
+Buffers have the following usages (and commands inducing that usage):
+ - `VERTEX` for the `buffers` in calls to `WebGPUCommandEncoder.setVertexBuffers`
+ - `INDEX` for `buffer` in calls to `WebGPUCommandEncoder.setIndexBuffer`
+ - `INDIRECT` for `indirectBuffer` in calls to `WebGPUCommandEncode.{draw|drawIndexed|dispatch}Indirect`
+ - `UBO`, `RO_STORAGE` and `STORAGE` for buffers referenced by bindgroups passed to `setBindGroup`, with the usage corresponding to the binding's type.
+ - `TRANSFER_SRC` for buffers used as the copy source of various commands.
+ - `TRANSFER_DST` for buffers used as the copy destination of various commands.
+ - (Maybe [RO_]STORAGE_TEXEL and UNIFORM_TEXEL?)
+
+Textures have the following usages:
+ - `OUTPUT_ATTACHMENT` for the subresources referenced by `WebGPURenderPassDescriptor`
+ - `SAMPLED`, `RO_STORAGE` and `STORAGE` for subresources corresponding to the image views referenced by bindgroups passed to `setBindGroup`, with the usage corresponding to the binding's type.
+ - `TRANSFER_SRC` for textures used as the copy source of various commands.
+ - `TRANSFER_DST` for textures used as the copy destination of various commands.
+ 
+Read only usages are `VERTEX`, `INDEX`, `INDIRECT`, `UBO`, `RO_STORAGE`, `TRANSFER_SRC` and `SAMPLED`.
+
+Note that here we split `RO_STORAGE` and `STORAGE` compared to other graphics APIs and that creating a resource with `STORAGE` implies `RO_STORAGE`.
```

Comment:
this needs more clarification, it's not totally clear from the sentence what "other graphics APIs" features are this being compared to

Reply:
Native graphics APIs don't have a flag for "read-only storage", only for "storage" so that's a new concept.